pkg/controller/upload-controller.go

```diff
@@ -126,7 +126,7 @@ func (r *UploadReconciler) Reconcile(_ context.Context, req reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 	// force cleanup if PVC pending delete and pod running or the upload/clone annotation was removed
-	if !shouldReconcile || podSucceededFromPVC(pvc) || pvc.DeletionTimestamp != nil {
+	if !shouldReconcile || (podSucceededFromPVC(pvc) && !isPodRetainAfterCompletion(pvc)) || pvc.DeletionTimestamp != nil {
```

Comment:
Again, the cleanup does more than just delete the pod, so we should let the cleanup run, but skip the pod deletion in there if the annotation is set.

Reply:
Sure.

Reply:
so we should still deleteService() even if we keep the pod, right?